src/shapes/radioInput.class.js

```diff
@@ -0,0 +1,162 @@
+       * @type Array
+       * @default
+       */
+      checked: undefined,
+      /**
+       * Points array
+       * @type Array
+       * @default
+       */
+      onChange: undefined,
+      /**
+       * Points array
+       * @type Array
+       * @default
+       */
+      onCheckCallback: undefined,
+      /**
+       * Points array
+       * @type Array
+       * @default
+       */
+      onUnCheckCallback: undefined,
+
+      /**
+       * Constructor
+       * @param {Array} points Array of points (where each point is an object with x and y)
+       * @param {Object} [options] Options object
+       * @return {fabric.RadioInput} thisArg
+       * @example
+       * var poly = new fabric.RadioInput([
+       *     { x: 10, y: 10 },
+       *     { x: 50, y: 30 },
+       *     { x: 40, y: 70 },
+       *     { x: 60, y: 50 },
+       *     { x: 100, y: 150 },
+       *     { x: 40, y: 100 }
+       *   ], {
+       *   stroke: 'red',
+       *   left: 100,
+       *   top: 100
+       * });
+       */
+
+      onCheckChange: function () {
+        if (this.isChecked() === true) {
+          this.set("fill", "transparent");
+          if (typeof this.onUnCheckCallback === "function") {
+            this.onUnCheckCallback(this);
+          }
+        } else {
+          this.set("fill", this.onCheckFillColor);
+          if (typeof this.onCheckCallback === "function") {
+            this.onCheckCallback(this);
+          }
+        }
+        if (typeof this.onChange === "function") {
+          this.onChange(this, this.checked);
+        }
+        this.canvas.renderAll();
+      },
+
+      onMouseDown: function (e) {
+        if (e.target === this) {
+          this.onCheckChange();
+          this.checked = !this.checked;
+        }
+      },
+      isChecked: function () {
+        return this.checked;
+      },
+
+      initialize: function (element, options) {
+        options = options || {};
+        var _this = this;
+        if (element.checked === true) {
+          element.fill = element.onCheckFillColor;
+        } else {
+          element.fill = "transparent";
+          element.checked = false;
+        }
+        this.callSuper("initialize", element, options);
+
+        this.on("added", function () {
+          this.canvas.on("mouse:down", function (e) {
+            _this.onMouseDown(e);
+          });
+        });
+      },
+
+      /**
+       * Returns object representation of an instance
+       * @param {Array} [propertiesToInclude] Any properties that you might want to additionally include in the output
+       * @return {Object} Object representation of an instance
+       */
+      toObject: function (propertiesToInclude) {
+        return fabric.util.object.extend(
+          this.callSuper(
+            "toObject",
+            [
+              "checked",
+              "onCheckFillColor",
+              "onChange",
+              "onCheckCallback",
+              "onUnCheckCallback",
```

Comment:
we do not support anymore serialization of functions. Is not secure and hard to grasp for developers.